src/bin/common.rs

```diff
@@ -126,10 +126,9 @@ pub fn process_frame(ctx: &mut Context,
                 _ => panic! ("unknown input bit depth!"),
             }
 
-            eprintln!("{}", ctx);
             let _ = ctx.send_frame(input);
-
             let pkt = ctx.receive_packet().unwrap();
+            eprintln!("{}", ctx);
             write_ivf_frame(output_file, pkt.number as u64, pkt.data.as_ref());
```

Comment:
At this point, ctx.fi.number and pkt.number differ by one but ctx.fi.frame_type matches the encoded packet.